### PR TITLE
testing/docker-compose: Fix missing deps

### PR DESCRIPTION
**Pull request by bratkartoffel**
This closes Bug #10282
https://bugs.alpinelinux.org/issues/10282

### CODE REVIEW

**Comment by tcely** [comment]
Please add running `docker-compose version` to check to catch this kind of problem in the future.

**Comment by tcely** [comment]
https://docs.docker.com/release-notes/docker-compose/

This can be upgraded to the latest version.

**Comment by tcely** [comment]
Moving docker-compose to community should help with CI failures.

**Comment by bratkartoffel** [comment]
I think we should not mix too many actions into one PR and focus on making it work again.
We can upgrade / move this aport later.

File: community/py-pyrsistent/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+check() {
+        cd "$builddir"
+        python3 setup.py test
+}
+
+package() {
+	mkdir -p "$pkgdir"
+}
+
+_py2() {
```
Comment (Ikke) [inline community/py-pyrsistent/APKBUILD]: Do you need python2 support here? Alpine LInux is going to drop python2 support in the near future. docker-compose is using python, so for that it's not needed.

Comment (bratkartoffel) [reply community/py-pyrsistent/APKBUILD]: You're right, I just added python2 support as the library provides it.
Will remove python2 support and rebase this evening.

**Comment by Ikke** [comment]
py3-pyrsistent has just been pushed (I didn't realize it was part of this PR), so it can be dropped from this PR.

Also notice that you're adding the packages directly to community, while they should start in testing (and as docker-compose is in testing, there is no need for them to go directly to community).

**Comment by tcely** [comment]
See https://github.com/alpinelinux/aports/pull/7233

**Comment by bratkartoffel** [comment]
> Also notice that you're adding the packages directly to community, while they should start in testing (and as docker-compose is in testing, there is no need for them to go directly to community).

py-pyrsistent is a dependency of community/py-jsonschema, so it also has to be in community, hasn't it?

File: community/py-pyrsistent/APKBUILD (hunk 1 above)
Comment (bratkartoffel) [reply community/py-pyrsistent/APKBUILD]: As this aport is a dependency of py-jsonschema this means that we have to drop python2 support there too. I'm unsure if we should do this now

**Comment by Ikke** [comment]
@bratkartoffel Ah, you are right, then I'll move pyrsistent to community first.

**Comment by Ikke** [comment]
py3-pyrsistent has been [moved][commit] to community

[commit]:https://github.com/alpinelinux/aports/commit/3b978a8

**Comment by tomalok** [comment]
It should also be noted that the latest release of `docker-compose` is at [1.24.0](https://github.com/docker/compose/releases/tag/1.24.0).

I fiddled around with trying to get the tests working a day or two ago -- got them to go further by adding `python3-dev` to `checkdepends="..."`.  After the test suite printed "running build_ext", it tried to execute `docker-compose test` (an unknown subcommand) and failed.

**Comment by mtneug** [comment]
Just a note: the dependencies are not only missing for `docker-compose`, but e.g. for `cloud-init` as well.

**Comment by Ikke** [comment]
Dependencies issues are fixed now. 

It appears you forgot to `apk checksum` after the last update.

**Comment by Ikke** [comment]
I fixed the remaining issues and pushed it. Note that I removed the simple `docker-compose -v` check, as:

1. Just running version does often not assert that it's actually working
2. Running `docker-compose` in the source directory is not even the same as the installed version. `docker-compose -v` ran just fine in the srcdir, but installed by setuptools, it complains about paramiko missing.

**Pull request closed without merging**